tests/model/test_model.py

```diff
@@ -57,3 +59,80 @@ def test_init(self, model, initial_data, user_profile):
             initial_data['unread_msgs']
         )
         assert model.unread_counts == []
+
+    def test_success_get_messages(self, mocker, messages_successful_response,
+                                  index_all_messages, initial_data):
+        # Initialize Model
+        mocker.patch('zulipterminal.model.Model.fetch_initial_data',
+                     return_value=initial_data)
+        mocker.patch('zulipterminal.model.Model.get_all_users',
+                     return_value=[])
+        mocker.patch('zulipterminal.model.Model.get_subscribed_streams',
+                     return_value=[])
+        self.classify_unread_counts = mocker.patch(
+            'zulipterminal.model.classify_unread_counts',
+            return_value=[])
+
+        # Setup mocks before calling get_messages
+        self.client.do_api_query.return_value = messages_successful_response
+        mocker.patch('zulipterminal.model.index_messages',
+                     return_value=index_all_messages)
+        model = Model(self.controller)
+        request = {
+            'anchor': model.anchor,
+            'num_before': model.num_before,
+            'num_after': model.num_after,
+            'apply_markdown': False,
+            'use_first_unread_anchor': True,
+            'client_gravatar': False,
+            'narrow': json.dumps(model.narrow),
+        }
+        model.client.do_api_query.assert_called_once_with(
+            request, '/json/messages', method="GET")
+        assert model.index == index_all_messages
+        anchor = messages_successful_response['anchor']
+        assert model.index[str(model.narrow)] == anchor
+        assert model.update is True
+
+        # TEST FOR get_messages() with first_anchor=False
```

Comment:
I would split these into separate tests - if and when a test fails, it would be easier to figure out which, and start debugging.